msal/tests/Test.MSAL.NET.Unit.net45/TelemetryTests.cs

```diff
@@ -411,5 +432,53 @@ public void AuthorityNotInTrustedHostList_AuthorityIsSetAsNullValueTest()
             }
             Assert.IsTrue(myReceiver.EventsReceived.Count > 0);
         }
+
+        [TestMethod]
+        [TestCategory("TelemetryInternalAPI")]
+        public void TelemetryEventCountsAreCorrectTest()
```

Comment:
Since you've written some concurrent code, have a think about a test that tries to make sure your code works in multi threaded env. For example, you could create 100 tasks that all create events, start them all up, wait for all for all of them to finish and do your counts.